### DISCUSSION TITLE
Finding Python types for Columns

### DISCUSSION THREAD

**Discussion by jenstroeger** [tiangolo/pydantic-sqlalchemy#97] [Questions] [answered]
Hi, thanks for this function just as I started putting together one of my own. Regarding this code:

https://github.com/tiangolo/pydantic-sqlalchemy/blob/8667e213a6ad8e62288fad2a6621958e8ab0b688/pydantic_sqlalchemy/main.py#L24-L30

Have you looked at the [typing.get_type_hints()](https://docs.python.org/3/library/typing.html#typing.get_type_hints) function? Perhaps 
```python
else:
    python_types = typing.get_type_hints(db_model)
    python_type = python_types[column.name]
```
works for you as well?

**Comment by ljluestc** [comment]
from sqlalchemy.sql import sqltypes
import typing

# ...

def python_type_for_column(db_model, column):
    # Check if the column type is in SQLAlchemy's type map
    if isinstance(column.type, sqltypes.TypeEngine):
        python_type = column.type.python_type
    else:
        python_types = typing.get_type_hints(db_model)
        python_type = python_types[column.name]
    return python_type

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :sweat_smile:

I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things. :rocket: 

That also means it uses all the nice type annotations utils from Pydantic. :nerd_face:
